TIL.md

```diff
@@ -0,0 +1,263 @@
+ 
+  ```` js
+  setupComponentTest('product-edit', {
+    integration: true
+  });
+
+  it('renders', function() {
+    this.set('product', { name: 'Phone', description: 'Black', sku: 1 });
+    this.render(hbs`{{product-edit product=product}}`);
+    expect(find('[data-test-product-form-name]').value).to.include('Phone');
+  });
+  ````
+## Acceptance test
+- integration tests ensure the application is working properly
+- without the use of page object
+  ```` js
+    ...
+    await visit('/products');
+    await click('[data-test-product-thumb-link]');
+    expect(find('[data-test-product-show-name]').textContent)
+      .to.include('Personal Computer');
+    await click('[data-test-products-link]');
+    ...
+  ````
+- with page object 
+  ```` javascript
+  // pages/product.js
+    import {
+      create,
+      visitable,
+      clickable,
+      text
+    } from 'ember-cli-page-object';
+
+    export default create({
+      visit: visitable('/products'),
+      openProduct: clickable('[data-test-product-thumb-link]'),
+      productName: text('[data-test-product-show-name]'),
+      openProducts: clickable('[data-test-products-link]')
+    });
+  ````
+
+  ```` javascript  
+  ...
+    await page.visit();
+    await page.openProduct();
+    expect(page.productName).to.include('Personal Computer');
+  ...
+  ````
+(we ensure the application is working properly, the process of transition from .. to ..)
+
+# Monday
+
+## Routing, template, components
+
+for picture, look in /TIL (ember.concepts.png)
+
+- router - is the best place to load data connected to the application path (/products - all products, /proudcts/1 - one product)
+- products router - loads data using model() hook (all products)
+- products template - we invoke component product-thumb for each element of products collection(available through model) with param product=product
+- product-thumb component - has access to product inside template.hbs and component.js (this.product)
+
+## Mirage
+- fake server that runs in client, useful in development, as with standard ember config of models and routes, it can be employed to simulate proper backend until it is implemented
+
+## Components
+- components can be generated using `ember g component --pod component-name`, pods are creating a neat structure of `tempate.hbs` and `component.js` in the `component-name` folder. Note: the name of a component has to contain a dash 
+
+# Thursday
+
+## Linters
+https://github.com/DockYard/eslint-plugin-ember-suave
+https://github.com/rwjblue/ember-cli-template-lint
+https://github.com/rwjblue/ember-template-lint
+- all in all - tools that are used to detect foul code style
+(cli-linter, eslinter, template-linter, suave)
+
+## Testing environment
+- qunit default was replaced with mocha (https://mochajs.org/ which is a test framework running on Node and in browser
```

Comment:
Add link and checkbox (that you can check after reading the documentation)
[ ] [mocha](https://mochajs.org/)
[ ] [chai](http://chaijs.com/api/bdd/)
[ ] [ember mocha](https://github.com/emberjs/ember-mocha/blob/master/README.md)